saito-core/src/core/data/storage.rs

```diff
@@ -136,10 +146,40 @@ impl Storage {
     //
     // token issuance functions below
     //
-    pub fn return_token_supply_slips_from_disk(&self) -> Vec<Slip> {
-        // let mut v: Vec<Slip> = vec![];
-        // let mut tokens_issued = 0;
+    pub async fn get_token_supply_slips_from_disk(&self) -> Vec<Slip> {
+        let mut v: Vec<Slip> = vec![];
+        let mut tokens_issued = 0;
         //
+        if self.file_exists(ISSUANCE_FILE_PATH).await {
+            if let Ok(lines) = self
+                .io_interface
+                .read_value(ISSUANCE_FILE_PATH.to_string())
+                .await
+            {
+                let mut contents = String::from_utf8(lines).expect("Failed to convert to String");
```

Comment:
for IO related things check the returned result rather than .expect(). better to print the exact error also